subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/JUnitPlatformTestEngineIntegrationTest.groovy

```diff
@@ -0,0 +1,323 @@
+            package org.gradle.other;
+
+            import org.junit.jupiter.api.Test;
+
+            public class AnotherTest {
+                @Test
+                void test() {
+                }
+            }
+        """
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            repositories {
+                mavenCentral()
+            }
+
+            testing {
+                suites {
+                    test {
+                        useJUnitJupiter {
+                            addEngine(JUnitPlatformSuiteEngine)
+                        }
+                        targets.all {
+                            testTask.configure {
+                                include '**/*Suite.class'
+                            }
+                        }
+                    }
+                }
+            }
+
+            import java.util.Collections
+            import org.gradle.api.plugins.jvm.testing.engines.JUnitPlatformTestEngine
+            import org.gradle.api.plugins.jvm.testing.engines.JUnitPlatformTestEngineParameters
+
+            abstract class JUnitPlatformSuiteEngine implements JUnitPlatformTestEngine<JUnitPlatformTestEngineParameters.None> {
+                Iterable<Dependency> getCompileOnlyDependencies() {
+                    return [ getDependencyFactory().create("org.junit.platform:junit-platform-suite-api:1.10.0") ]
+                }
+
+                Iterable<Dependency> getRuntimeOnlyDependencies() {
+                    return [ getDependencyFactory().create("org.junit.platform:junit-platform-suite-engine:1.10.0") ]
+                }
+            }
+        """
+
+        when:
+        succeeds("test")
+
+        then:
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.assertTestClassesExecuted("org.gradle.suite.OneTest")
+        result.testClassByHtml("org.gradle.suite.OneTest").assertTestCount(1, 0, 0)
+    }
+
+    def "can specify a custom test engine with parameters"() {
+        given:
+        file("src/test/java/org/gradle/OneTest.java") << """
+            package org.gradle;
+
+            import org.junit.Test;
+
+            public class OneTest {
+                @Test
+                public void test() {
+                }
+            }
+        """
+        file("src/test/java/org/gradle/AnotherTest.java") << """
+            package org.gradle;
+
+            import org.junit.Test;
+
+            public class AnotherTest {
+                @Test
+                public void test() {
+                }
+            }
+        """
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            repositories {
+                mavenCentral()
+            }
+
+            testing {
+                suites {
+                    test {
+                        useJUnitPlatform {
+                            addEngine(JUnitVintageEngine) {
```

Comment:
🤔 While looking at this, I thought that the it would read better if we called it `withEngine` instead of `addEngine`, then it reads like "use JUnit Platform with Engine JUnitVintageEngine"

Reply:
Sounds fine to me